src/SimpleGraphs/generators/randgraphs.jl

```diff
@@ -989,6 +989,144 @@ function uniform_tree(n::Integer; rng::Union{Nothing,AbstractRNG}=nothing)
     return prufer_decode(random_code)
 end
 
+"""
+    havel_hakimi_graph_generator(degree_sequence::AbstractVector{<:Integer})
+
+Returns a simple graph with a given finite degree sequence of non-negative integers generated via the Havel-Hakimi algorithm which works as follows:
+1. successively connect the node of highest degree to other nodes of highest degree;
+2. sort the remaining nodes by degree in decreasing order;
+3. repeat the procedure.
+
+## References
+1. [Hakimi (1962)](https://doi.org/10.1137/0110037);
+2. [Wikipedia](https://en.wikipedia.org/wiki/Havel%E2%80%93Hakimi_algorithm).
+"""
+function havel_hakimi_graph_generator(degree_sequence::AbstractVector{<:Integer})
+    # Check whether the degree sequence has only non-negative values
+    all(degree_sequence .>= 0) ||
+        throw(ArgumentError("The degree sequence must contain non-negative integers only."))
+    # Instantiate an empty simple graph 
+    graph = SimpleGraph(length(degree_sequence))
+    # Create a (vertex, degree) ordered dictionary
+    vertices_degrees_dict = OrderedDict(
```

Comment:
As long as it work correctly, we can keep it that way, but we probably could make this function much faster, by using a `Vector` of tuples instead. The sorting step should also be fairly easy to speed up, as after adding a vertex, the degree order is only slightly not sorted.

Reply:
We unfortunately don't have much time for this, but it would surely help. Would even be better if the final instantiation of a graph object was optional or put in a wrapper method, so that also other packages in the ecosystem may benefit from the full performance of the method (e.g. MutlilayerGraphs.jl's [configuration model-like constructors](https://juliagraphs.org/MultilayerGraphs.jl/stable/API/#MultilayerGraphs.MultilayerGraph-Union{Tuple{H},%20Tuple{U},%20Tuple{T},%20Tuple{Vector{%3C:Layer{T,%20U,%20G}%20where%20G%3C:AbstractGraph{T}},%20Vector{%3C:Interlayer{T,%20U,%20G}%20where%20G%3C:AbstractGraph{T}},%20UnivariateDistribution}}%20where%20{T%3C:Integer,%20U%3C:Real,%20H%3C:AbstractGraph{T}})).